src/seamless_communication/cli/m4t/finetune/trainer.py

```diff
@@ -351,53 +360,82 @@ def _train_step_log(self) -> None:
                 f"last lr={self.lr_scheduler.get_last_lr()[0]:.2E}"
             )
 
-    def _train_step(self, batch: dataloader.MultimodalSeqsBatch) -> None:
+    def _train_forward(self, batch: List[dataloader.MultimodalSeqsBatch]) -> None:
```

Comment:
why do you call it forward? we do a full cycle: forward, backward, optimizer-step here

Reply:
Just for lack of a better word, no good reason. Because "step" is used in the eval function to iterate over all the batches whereas here it is used to run 1 batch.